### ISSUE TITLE
ExtJS 6.2.1

### ISSUE THREAD

**Issue by FrenchCruzian** [12ftguru/extjs-ux-mediator#3]
Mediator publish
Mediato...7170820 (line 26)
TypeError: this.fireEventArgs is not a function
return this.fireEventArgs(eventName, arraySlice.call(arguments, 1));

I followed this tutorial:
https://medium.com/twelve-foot-guru/extjs-6-getting-your-controllers-to-talk-to-one-another-863189cc1d56

**Comment by Zekkenn** [comment]
Hello! I know this is an old post, but did you fix it? 
Currently, i'm facing the same problem.

**Comment by bpjohnson** [comment]
No. Generally, if this.fireEventArgs is not a function, then "this" does not implement or extend Ext.util.Observable, so cannot fire events, thus can't be a mediator.

Additionally, there are built-in ways to solve the same problem now, that I would recommend you investigate, rather than use this old code.

Specifically https://docs.sencha.com/extjs/6.6.0/classic/Ext.app.EventDomain.html

Which you can use to set up listeners that listen to events from all controllers. (See the "listen" config on Ext.app.ViewController. The part that talks about " Ext.app.domain.Controller domain can be used to listen to events fired by other Controllers".

https://docs.sencha.com/extjs/6.6.0/classic/Ext.app.ViewController.html#cfg-listen
